From 190d8350868dbdb45b61d9d58d293b03dc790b0b Mon Sep 17 00:00:00 2001
From: Hector Oron <zumbi@debian.org>
Date: Tue, 10 May 2011 15:49:34 -0300
Subject: [PATCH] config.guess: add support for ARM abihf

2011-05-09  Hector Oron <zumbi@debian.org>
            Konstantinos Margaritis <markos@debian.org>

       * config.guess: Add configuration support for
       ${UNAME_MACHINE}-unknown-linux-gnueabihf on arm targets.
---
 ChangeLog    |    6 ++++++
 config.guess |    8 +++++++-
 2 files changed, 13 insertions(+), 1 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index c76e2f3..58140e2 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,9 @@
+2011-05-09  Hector Oron <zumbi@debian.org>
+	    Konstantinos Margaritis <markos@debian.org>
+
+	* config.guess: Add configuration support for
+	${UNAME_MACHINE}-unknown-linux-gnueabihf on arm targets.
+
 2011-03-31  Kirk Hays  <khays@hayshaus.com>
 	    Ben Elliston  <bje@gnu.org>
 
diff --git a/config.guess b/config.guess
index 187cd54..378e703 100755
--- a/config.guess
+++ b/config.guess
@@ -882,7 +882,13 @@ EOF
 	then
 	    echo ${UNAME_MACHINE}-unknown-linux-gnu
 	else
-	    echo ${UNAME_MACHINE}-unknown-linux-gnueabi
+	    if echo __ARM_PCS_VFP | $CC_FOR_BUILD -E - 2>/dev/null \
+	        | grep -q __ARM_PCS_VFP
+	    then
+	        echo ${UNAME_MACHINE}-unknown-linux-gnueabi
+	    else
+	        echo ${UNAME_MACHINE}-unknown-linux-gnueabihf
+	    fi
 	fi
 	exit ;;
     avr32*:Linux:*:*)
-- 
1.7.2.5

